### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by Jeffrey-Vervoort-KNMI**
Adds tests for:
- Deleting a timeseries with observations
- Finding a timeseries via a Polygon

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -77,6 +77,8 @@ services:
     depends_on:
       store:
         condition: service_healthy
+#      loader:
+#        condition: service_completed_successfully
```
Comment (lukas-phaf) [inline docker-compose.yml:81]: Not needed, as they are not automatically started.

File: integration-test/test_dummy_data.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
+# Note that this assumes that the KNMI test data is loader (using loader container)
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:1]: Does it?

File: integration-test/test_dummy_data.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import logging
+import os
+from datetime import datetime
+from datetime import timezone
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:5]: I am not a fan of the file name. Maybe better to split it up in two?
`test_delete`
`test_polygon`

Also, why not test the polygon lookup on the actual KNMI data (like the lookup by station and parameter)? It saves a lot of setup code.

File: integration-test/test_dummy_data.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+@pytest.fixture(scope="session")
+def grpc_stub():
+    with grpc.insecure_channel(f"{os.getenv('DSHOST', 'localhost')}:{os.getenv('DSPORT', '50050')}") as channel:
+        yield dstore_grpc.DatastoreStub(channel)
+
+
+@pytest.fixture(scope="function")
+def dummy_timeseries_for_delete():
+    dummy_id = 999999999
+    dummy_str_id = "999999999"
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:27]: Call this `test_delete-1` for easy identification if it ever leaks.

File: integration-test/test_dummy_data.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    ]
+    obs_put_request = dstore.PutObsRequest(
+        tsobs=[dstore.TSObservations(tsid=dummy_id, obs=obs)],
+    )
+    return obs_put_request
+
+
+def test_delete_timeseries(grpc_stub, dummy_timeseries_for_delete, dummy_observations_for_delete):
+    ts_add_response = grpc_stub.AddTimeSeries(dummy_timeseries_for_delete)
+    assert str(ts_add_response) == "status: -1\n"
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:68]: Let's not hardcode these "bogus" response codes in the test.

File: integration-test/test_dummy_data.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        other2="test_value5",
+        other3="test_value6",
+    )
+    ts_add_request_2 = dstore.AddTSRequest(
+        id=dummy_id,
+        metadata=ts_metadata,
+    )
+    grpc_stub.AddTimeSeries(ts_add_request_2)
+
+    yield  # Hereafter teardown
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:138]: Nice!

File: integration-test/test_dummy_data.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+    obs_get_response = grpc_stub.GetObservations(obs_get_request)
+    assert obs_get_response.tsobs[0].tsid == dummy_timeseries_for_delete.id
+    assert len(obs_get_response.tsobs[0].obs) == 0
+
+
+@pytest.fixture(scope="module")
+def setup_dummy_data_for_polygon(grpc_stub):
+    dummy_id = 999999990
+    dummy_str_id = "999999990"
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:107]: If we keep this, give it an identifiable name.

File: integration-test/test_dummy_data.py [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+        metadata=ts_metadata,
+    )
+    grpc_stub.AddTimeSeries(ts_add_request_1)
+
+    dummy_id = 999999991
+    dummy_str_id = "999999991"
+    ts_metadata = dstore.TSMetadata(
+        station_id=dummy_str_id,
+        param_id=dummy_str_id,
+        pos=dstore.Point(lat=90, lon=180),
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:127]: Although an interesting case to break the libary, I would move these points away from the north pole and the dateline.

File: integration-test/test_dummy_data.py [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+        # Middle bottom, should fall outside since polygon is curved because the earth is round (postgres geography).
+        ((85, 120), (80, 120), (85, -120), (80, -120)),
+        {"number_of_stations": 0, "station_ids": []},
+    ),
+    (
+        # Complex polygon
+        ((79, 169), (81, 169), (80, 171), (89, 171), (89, 180), (75, 180), (75, 171)),
+        {"number_of_stations": 1, "station_ids": [999999990]},
+    ),
+]
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:174]: Rewrite the tests away from the pole and the dateline. I am not sure these test what they say they do.
For example, (90, 180) == (80, 180).

File: integration-test/test_dummy_data.py [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+        {"number_of_stations": 0, "station_ids": []},
+    ),
+    (
+        # middle top
+        ((85, 120), (90, 120), (90, -120), (85, -120)),
+        {"number_of_stations": 1, "station_ids": [999999991]},
+    ),
+    (
+        # Middle bottom, should fall outside since polygon is curved because the earth is round (postgres geography).
+        ((85, 120), (80, 120), (85, -120), (80, -120)),
```
Comment (lukas-phaf) [inline integration-test/test_dummy_data.py:166]: This polygon it a figure-8, or am I missing something?

File: integration-test/test_dummy_data.py (hunk 2 above)
Comment (Jeffrey-Vervoort-KNMI) [reply integration-test/test_dummy_data.py:1]: It does not, I used the other file as starting point, simply copied it and did not notice the top comment.

File: docker-compose.yml (hunk 1 above)
Comment (Jeffrey-Vervoort-KNMI) [reply docker-compose.yml:81]: It depends, if you never remove your volume it is unnecessary. If you do, your integration tests will fail when running `docker compose --profile test down --volumes && docker compose --profile test up -d`.

Comment (lukas-phaf) [reply docker-compose.yml:81]: The profile is not intended to be used together with `up`. Try this instead
`docker compose down --volumes && docker compose --profile test build && docker compose up -d && docker compose run --rm loader && docker compose run --rm integration`

(copied from what the github workflow does).

File: integration-test/test_knmi.py [hunk 11]
```diff
@@ -73,3 +62,71 @@ def test_get_values_single_station_single_paramters(grpc_stub):
...
+    ),
+]
+
+
+@pytest.mark.parametrize("coords,expected", input_params_polygon)
+def test_get_observations_with_polygon(grpc_stub, coords, expected):
+    ts_request = dstore.FindTSRequest(
+        inside=dstore.Polygon(points=[dstore.Point(lat=lat, lon=lon) for lat, lon in coords])
+    )
+    ts_response = grpc_stub.FindTimeSeries(ts_request)
```
Comment (lukas-phaf) [inline integration-test/test_knmi.py:127]: I suggest to add a single parameter (for example "rh") to the query to lower the maintenance of the test (if the number of time-series changes).

**Pull request merged**